tensorly/utils/tests/test_jointdiag.py

```diff
@@ -0,0 +1,39 @@
+import numpy as np
+import tensorly as tl
+from ..jointdiag import jointdiag
+
+
+def test_jointdiag():
+    """
+    Generates random diagonal tensor, and random mixing matrix
+    Multiplies every slice of diagonal tensor with matrix 'synthetic' S * D * S^-1
+    Sends altered tensor into jointdiag function
+    Returns the diagonal tensor estimate and mixing matrix estimate
+    """
+    k = 14
+    d = 10
+
+    rng = np.random.RandomState()
+    mixing = rng.randn(d, d)
+    diags = np.zeros((d, d, k))
+    synthetic = np.zeros((d, d, k))
+
+    # Generates k random diagonal matrices and scrambles according to the mixing matrix
+    for i in range(k):
+        temp_diag = np.diag(rng.randn(d))
+        diags[:, :, i] = temp_diag
+        synthetic[:, :, i] = np.linalg.inv(mixing) @ temp_diag @ mixing
```

Comment:
This is actually quite useful to understand the type of mixing allowed: might be useful to have a short Notes section in the doctoring to explain what kind of scrambling the algorithm supports and roughly what it does?